x-pack/plugin/sql/src/main/antlr/SqlBase.g4

```diff
@@ -163,14 +163,18 @@ expression
 booleanExpression
     : NOT booleanExpression                                                                 #logicalNot
     | EXISTS '(' query ')'                                                                  #exists
-    | QUERY '(' queryString=string (',' options=string)* ')'                                #stringQuery
-    | MATCH '(' singleField=qualifiedName ',' queryString=string (',' options=string)* ')'  #matchQuery
-    | MATCH '(' multiFields=string ',' queryString=string (',' options=string)* ')'         #multiMatchQuery
+    | QUERY '(' queryString=string matchQueryOptions ')'                                    #stringQuery
+    | MATCH '(' singleField=qualifiedName ',' queryString=string matchQueryOptions ')'      #matchQuery
+    | MATCH '(' multiFields=string ',' queryString=string matchQueryOptions ')'             #multiMatchQuery
     | predicated                                                                            #booleanDefault
     | left=booleanExpression operator=AND right=booleanExpression                           #logicalBinary
     | left=booleanExpression operator=OR right=booleanExpression                            #logicalBinary
     ;
 
+matchQueryOptions
```

Comment:
I chose to use this separate marker to avoid confusion in the `ExpressionBuilder` with `string()` containing also the `queryString` argument which had to be skipped.

Reply:
Is it possible to not have it as a top-level declaration and use something like `options=(',' string)*` instead?

Reply:
Nope, unfortunately not, because of the `*` wildcard.